### ISSUE TITLE
Adding bubble scroll view to the app

### ISSUE THREAD

**Issue by madhurgupta10** [commons-app/apps-android-commons#2590]
**Summary:** 
Bubble scroll is popular among apps and helps the user to easily locate the photo they are scrolling for.

**Screen-shots:** 
![](https://camo.githubusercontent.com/26b1a68095f8cf1e68add244bdcea7fdc376a341/687474703a2f2f692e696d6775722e636f6d2f556771687a75642e676966?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=4156)

**Would you like to work on the issue?**
No

**Comment by nicolas-raoul** [comment]
Does this work well with lists whose size increment as you scroll?

For what activity do you want to implement this?

**Comment by madhurgupta10** [comment]
@nicolas-raoul I was thinking about implementing it in in Contributions activity, and yes it works well with different sized list.

**Comment by nicolas-raoul** [comment]
In the future we will make the Contributions activity load only a dozen elements, and load more as the user scrolls down.

Please give a few examples of applications that have bubble scroll on a list whose elements are automatically added as you scroll down. That will allow us to study their behaviors and choices.

Thanks! :-)

**Comment by madhurgupta10** [comment]
@nicolas-raoul, for example, we can consider google photos app, where the bubble scroll view features the dates, helps user to navigate easily and finds the photos they need. But in case we are adding infinite scrolling feature then we might need to refresh the view.

**Comment by albendz** [comment]
My 2 cents:

For dynamic loading, you may be able to create all lettered sections in a top level view and each lettered view has its own set of scrollable data.. If you are within one section of the listview, you dynamically load x items within that section on scroll. When you are returned fewer than x items for that section, you begin loading the next section's data into the next section. It would be sort of like having a nested scroll view where the top level is the alphabetical segments and the inner view are the loaded pieces of data. I apologize if that's confusing.

Another point to consider is how this will behave in other languages. It will be tricky to make this language independent or force all languages to be compatible with a specific alphabet.

**Comment by nicolas-raoul** [comment]
If we are talking about the Contributions activity, I guess the bubble would not show letters, right @madhurgupta10 ? What would the bubbles show instead? Upload dates?

**Comment by madhurgupta10** [comment]
@nicolas-raoul Yes, I was thinking about displaying dates, just like google photos.

**Comment by VinamraGuptaa** [comment]
Is this issue still open?

**Comment by nicolas-raoul** [comment]
@VinamraGuptaa Yes it is open. This is labeled `enhancement` though. Please make sure to read the "more than 5 previous pull requests" part at https://github.com/commons-app/commons-app-documentation/blob/master/android/Volunteers-welcome!.md Thanks!
